### ISSUE TITLE
YouTube playback not working with Waterfox Classic 2019.10 (20191016181224) on Windows

### ISSUE THREAD

**Issue by grahamperrin** [MrAlex94/Waterfox#1205]
Mike Diack https://www.waterfox.net/blog/waterfox-2019.10-release-download/#commento-comment-card-524e7248f39dcab385d235b3d03f69880e033c80b2d111501866a6c6a2a2965a and others at https://www.waterfox.net/blog/waterfox-2019.10-release-download/#commento reported YouTube not working. 

A shot of 20191016181224 in a (paused) virtual machine, Windows 10

![image](https://user-images.githubusercontent.com/192271/66964668-356eca00-f06f-11e9-9d8c-79932ec6b629.png)

Meta, tracking: #538

**Comment by Squall-Leonhart** [comment]
This is a Youtube service change.

**Comment by grahamperrin** [comment]
Reproducible with Waterfox Classic 2019.10 (20191015002329) on Manjaro. 

Reproducible with Waterfox Classic 56.2.14 (20190903144949) on Manjaro at https://www.youtube.com/watch?v=Z4qF1uisE04: 

![image](https://user-images.githubusercontent.com/192271/66969763-d3b85b00-f082-11e9-810c-64c2f97be1c4.png)

**Comment by grahamperrin** [comment]
**Not** reproducible with Waterfox Classic 56.2.14 (20191007063316) on FreeBSD-CURRENT: 

![image](https://user-images.githubusercontent.com/192271/66969418-52ac9400-f081-11e9-89db-813d087cfabe.png)

**Comment by grahamperrin** [comment]
> This is a Youtube service change.

In the Firefox area, https://redd.it/diycr9 an hour ago appears to involve **polymer** (#1206), not playback, but that's a lazy search result. 

For now (early hours, 03:00) I'd like to keep the two Waterfox issues separate, wait/sleep and see how the dusts settle …

**Comment by Squall-Leonhart** [comment]
UI is back to normal, hows playback for you Graham?

**Comment by grahamperrin** [comment]
Thanks, OK, I _might_ have FreeBSD bug [233301 – emulators/virtualbox-ose - multimedia issues on 13-current](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=233301), which I had in the past _but not_ when I last tested AV in a guest (IIRC I tested Amazon Video with DRM, and playback did succeed). 

I do have a home-built installation: 

```text
$ date ; uname -v
Thu 17 Oct 2019 06:28:16 BST
FreeBSD 13.0-CURRENT #33 r352942: Wed Oct  2 05:26:54 BST 2019     root@momh167-gjp4-8570p:/usr/obj/usr/src/amd64.amd64/sys/GENERIC-NODEBUG 
$ pkg search -r FreeBSD virtualbox-ose
virtualbox-ose-additions-5.2.32_1 VirtualBox additions for FreeBSD guests
virtualbox-ose-additions-nox11-5.2.32_1 VirtualBox additions for FreeBSD guests
virtualbox-ose-kmod-5.2.32_1   VirtualBox kernel module for FreeBSD
$ pkg query '%o %v %R' virtualbox-ose virtualbox-ose-kmod
emulators/virtualbox-ose 5.2.32_1 poudriere
emulators/virtualbox-ose-kmod 5.2.32_1 poudriere
$ pkg info virtualbox-ose | grep -i installed
Installed on   : Thu Oct 10 07:06:48 2019 BST
$ 
```

– however now I see, there's no longer the `Makefile.local` that was mentioned at <https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=233301#c8>: 

```text
root@momh167-gjp4-8570p:/usr/local/poudriere/ports/default # cat emulators/virtualbox-ose/Makefile.local
cat: emulators/virtualbox-ose/Makefile.local: No such file or directory
root@momh167-gjp4-8570p:/usr/local/poudriere/ports/default # 
```

https://svnweb.freebsd.org/ports?view=revision&revision=514643 a minor update was ported around half an hour ago, it'll take a few hours for me to build everything that's required (with or without `audio/pulseaudio`) …

**Comment by grahamperrin** [comment]
tl;dr apparently not an issue with VirtualBox. YouTube maybe pernickety at the moment. I have not yet got https://www.youtube.com/watch?v=kSVaBXkYges to play …

**Comment by grahamperrin** [comment]
Windows 10 in VirtualBox, https://www.youtube.com/watch?v=kSVaBXkYges

- plays in Firefox 68.1.0esr
- plays in Waterfox Current 2019.10 (20191011232540)
- still does not play in Waterfox Classic 2019.10 (20191016181224)

![image](https://user-images.githubusercontent.com/192271/67115155-741e9480-f1d5-11e9-840f-898948e20fbc.png)

Side note: the license for this installation of Windows will expire in around sixteen hours.

**Comment by grahamperrin** [comment]
Manjaro guest VM: non-conclusive result. No playback in Firefox, either: 

![image](https://user-images.githubusercontent.com/192271/67116647-c1e8cc00-f1d8-11e9-99f9-dba821dff887.png)

Lubuntu and (outdated) GhostBSD guests: here, too, I can't get playback in Firefox. This is, notably, whilst **pushing the host machine quite hard** with coincidental build of part of the OS: 

![image](https://user-images.githubusercontent.com/192271/67117631-155c1980-f1db-11e9-84c1-a9f491a753f9.png)

These two VM states are saved, I ought to revisit each one when the system is less busy. 

Not forgetting FreeBSD bug 233301 affecting VirtualBox …

Note to self: 

```text
root@momh167-gjp4-8570p:/usr/ports/graphics/drm-legacy-kmod # make configure
===>  License BSD2CLAUSE MIT GPLv2 accepted by the user
===>   drm-legacy-kmod-g20190910 depends on file: /usr/local/sbin/pkg - found
===> Fetching all distfiles required by drm-legacy-kmod-g20190910 for building
===>  Extracting for drm-legacy-kmod-g20190910
=> SHA256 Checksum OK for FreeBSDDesktop-drm-legacy-g20190910-f5db697_GH0.tar.gz.
===>  Patching for drm-legacy-kmod-g20190910
===>  Configuring for drm-legacy-kmod-g20190910
root@momh167-gjp4-8570p:/usr/ports/graphics/drm-legacy-kmod # cd /usr/src
root@momh167-gjp4-8570p:/usr/src # make -j2 buildworld && make -j2 buildkernel KERNCONF=GENERIC-NODEBUG
```

**Comment by grahamperrin** [comment]
@hawkeye116477 should I continue to test here? 

Or am I likely _sometimes_ encountering the issues that #1203 intends to fix?

**Comment by hawkeye116477** [comment]
@grahamperrin In case of Windows, you should rather recompile with fix from #1203 and then test.

**Comment by grahamperrin** [comment]
> https://svnweb.freebsd.org/ports?view=revision&revision=514643 a minor update was ported around half an hour ago, it'll take a few hours for me to build everything that's required (with or without audio/pulseaudio) …

I built VirtualBox 5.2.34 with the workaround for FreeBSD bug 233301: 

```text
$ cat /usr/local/poudriere/ports/default/emulators/virtualbox-ose/Makefile.local 
PULSEAUDIO_CONFIGURE_OFF= 
BUILD_DEPENDS+= pulseaudio:audio/pulseaudio
$ 
```

– then used the built package to update my installation, then changed each virtual machine from _OSS Audio Driver_ to _PulseAudio_. 

Test case https://www.youtube.com/watch?v=kSVaBXkYges now plays without difficulty on Manjaro with: 

- Waterfox Classic 56.2.14 (20190903144949)
- Waterfox Classic 2019.10 (20191015002329)
- Waterfox Current 2019.10 (20191015121210).

On Windows 10, playback is fine with: 

- Firefox ESR 60.9.0 (20190901094603)
- Firefox ESR 68.1.0 (20190826132627)
- Waterfox Current 2019.10 (20191011232540).

**Comment by grahamperrin** [comment]
Not reproducible with build 20191022125504 of 2019.10 a.k.a. **2019.10-classic-1** for Windows.
